src/SBSpergel.cpp

```diff
@@ -0,0 +1,715 @@
+           <<func(solver.getUpperBound())<<std::endl;
+        solver.setMethod(Brent);
+        double r0 = solver.root();
+        dbg<<"Root is "<<r0<<std::endl;
+        return r0;
+    }
+
+    double SpergelInfo::getXNorm() const
+    { return std::pow(2., -_nu) / _gamma_nup1 / (2.0 * M_PI) / getFluxFraction(); }
+
+    double SpergelInfo::xValue(double r) const
+    {
+        if (_truncated && r > _trunc) return 0.;
+        else if (r == 0.) return _xnorm0;
+        else return boost::math::cyl_bessel_k(_nu, r) * std::pow(r, _nu);
+    }
+
+    double SpergelInfo::kValue(double ksq) const
+    {
+        if (_truncated) {
+            if (_ft.size() == 0) buildFT();
+            double lk=0.5*std::log(ksq);
+            if (lk < _a1) {
+                //linearly interpolate the first bin (in ksq)
+                return (1. - ksq/_a1ksq * (1. - _fta1/_a1ksq));
+            }
+            else if (lk > _ft.argMax()) return 0.;
+            else return _ft(lk)/ksq;
+        } else {
+            return std::pow(1. + ksq, -1. - _nu);
+        }
+    }
+
+    class SpergelIntegrand : public std::unary_function<double, double>
+    {
+    public:
+        SpergelIntegrand(double nu, double k) :
+            _nu(nu), _k(k) {}
+        double operator()(double r) const
+        { return std::pow(r, _nu)*boost::math::cyl_bessel_k(_nu, r) * r*j0(_k*r); }
+
+    private:
+        double _nu;
+        double _k;
+    };
+
+    void SpergelInfo::buildFT() const
+    {
+        assert(_trunc > 0.);
+        if (_ft.size() > 0) return;
+        dbg<<"Building truncated Spergel Hankel transform"<<std::endl;
+        dbg<<"nu = "<<_nu<<std::endl;
+        dbg<<"trunc = "<<_trunc<<std::endl;
+        // Do a Hankel transform and store the results in a lookup table.
+        double prefactor = std::pow(2., -_nu) / _gamma_nup1 / _flux;
+        dbg<<"prefactor = "<<prefactor<<std::endl;
+
+        // Along the way, find the last k that has a kValue > 1.e-3
+        double maxk_val = this->_gsparams->maxk_threshold;
+        dbg<<"Looking for maxk_val = "<<maxk_val<<std::endl;
+        // Keep going until at least 5 in a row have kvalues below kvalue_accuracy.
+        // (It's oscillatory, so want to make sure not to stop at a zero crossing.)
+
+        // We use a cubic spline for the interpolation, which has an error of O(h^4) max(f'''').
+        // I have no idea what range the fourth derivative can take for the hankel transform,
+        // so let's take the completely arbitrary value of 10.  (This value was found to be
+        // conservative for Sersic, but I haven't investigated here.)
+        // 10 h^4 <= kvalue_accuracy
+        // h = (kvalue_accuracy/10)^0.25
+        double dlogk = _gsparams->table_spacing * sqrt(sqrt(_gsparams->kvalue_accuracy / 10.));
+        dbg<<"Using dlogk = "<<dlogk<<std::endl;
+        int n_below_thresh = 0;
+
+        // Don't go past k = 500
+        double kmin = dlogk; // have to begin somewhere...
```

Comment:
I'm a little concerned about this.  It would be nice if there was a calculation to show that the linear approximation you use for k < kmin has errors less than kvalue_accuracy.  We usually try to make sure of that, and it's not obvious to me that this value for kmin satisfies that criterion.  

However, I've also proposed getting rid of the truncation option, which means this whole function would go away.  In which case the comment is moot.